src/gf.c

```diff
@@ -3164,15 +3164,14 @@ static uint64_t inference_start_time = 0;
 JL_DLLEXPORT void jl_typeinf_begin(void)
 {
     JL_LOCK(&typeinf_lock);
-    if (jl_measure_compile_time[jl_threadid()])
+    if (jl_atomic_load(&jl_measure_compile_time))
         inference_start_time = jl_hrtime();
 }
 
 JL_DLLEXPORT void jl_typeinf_end(void)
 {
-    int tid = jl_threadid();
-    if (typeinf_lock.count == 1 && jl_measure_compile_time[tid])
-        jl_cumulative_compile_time[tid] += (jl_hrtime() - inference_start_time);
+    if (typeinf_lock.count == 1 && jl_atomic_load(&jl_measure_compile_time))
```

Comment:
Same here–this read should only happen once (in jl_typeinf_begin). We can use `inference_start_time == 0` here as a flag to test whether it was true on entry.

Reply:
Thanks Jameson - fixed in latest push.